ui/button.js

```diff
@@ -30,389 +30,317 @@
 	defaultElement: "<button>",
 	options: {
 		disabled: null,
-		text: true,
+		showLabel: true,
 		label: null,
-		icons: {
-			primary: null,
-			secondary: null
+		icon: null,
+		iconPosition: "beginning",
+		classes: {
+			"ui-button": "ui-corner-all"
 		}
 	},
-	_create: function() {
-		this.element.closest( "form" )
-			.unbind( "reset" + this.eventNamespace )
-			.bind( "reset" + this.eventNamespace, formResetHandler );
 
-		if ( typeof this.options.disabled !== "boolean" ) {
-			this.options.disabled = !!this.element.prop( "disabled" );
-		} else {
-			this.element.prop( "disabled", this.options.disabled );
-		}
+	_getCreateOptions: function() {
+		var disabled,
+			options = this._super() || {};
 
-		this._determineButtonType();
-		this.hasTitle = !!this.buttonElement.attr( "title" );
+		this.isInput = this.element.is( "input" );
 
-		var that = this,
-			options = this.options,
-			toggleButton = this.type === "checkbox" || this.type === "radio",
-			activeClass = !toggleButton ? "ui-state-active" : "";
-
-		if ( options.label === null ) {
-			options.label = (this.type === "input" ? this.buttonElement.val() : this.buttonElement.html());
+		disabled = this.element[ 0 ].disabled;
+		if ( disabled != null ) {
+			options.disabled = disabled;
 		}
 
-		this._hoverable( this.buttonElement );
+		this.originalLabel = this.isInput ? this.element.val() : this.element.html();
+		if ( this.originalLabel ) {
+			options.label = this.originalLabel;
+		}
 
-		this.buttonElement
-			.addClass( baseClasses )
-			.attr( "role", "button" )
-			.bind( "mouseenter" + this.eventNamespace, function() {
-				if ( options.disabled ) {
-					return;
-				}
-				if ( this === lastActive ) {
-					$( this ).addClass( "ui-state-active" );
-				}
-			})
-			.bind( "mouseleave" + this.eventNamespace, function() {
-				if ( options.disabled ) {
-					return;
-				}
-				$( this ).removeClass( activeClass );
-			})
-			.bind( "click" + this.eventNamespace, function( event ) {
-				if ( options.disabled ) {
-					event.preventDefault();
-					event.stopImmediatePropagation();
-				}
-			});
-
-		// Can't use _focusable() because the element that receives focus
-		// and the element that gets the ui-state-focus class are different
-		this._on({
-			focus: function() {
-				this.buttonElement.addClass( "ui-state-focus" );
-			},
-			blur: function() {
-				this.buttonElement.removeClass( "ui-state-focus" );
-			}
-		});
+		return options;
+	},
 
-		if ( toggleButton ) {
-			this.element.bind( "change" + this.eventNamespace, function() {
-				that.refresh();
-			});
+	_create: function() {
+		this.formElement = $( this.element[ 0 ].form );
+
+		// We don't use _on and _off here because we want all the buttons in the same form to use
+		// single handler which handles all the button widgets in the form
+		this.formElement.off( "reset." + this.widgetFullName, formResetHandler );
+		this.formElement.on( "reset." + this.widgetFullName, formResetHandler );
+		buttonCount++;
```

Comment:
Since we're counting instances, use the counter here as well:

```js
// We don't use _on here, because we want all the buttons in the same form to use
// single handler which handles all the button widgets in the form
if ( buttonCount === 0 ) {
	this.formElement.on( "reset." + this.widgetFullName, formResetHandler );
}
buttonCount++;
```